src/pds_doi_service/core/actions/reserve.py

```diff
@@ -237,8 +243,24 @@ def run(self, **kwargs):
                 # Note that for both OSTI and DataCite, reserve requests should
                 # utilize the POST method
                 if not self._dry_run:
+                    service_type = DOIServiceFactory.get_service_type()
+
+                    # If a DOI has already been assigned by DataCite,
+                    # we need to use a PUT request on the URL associated to the DOI
+                    if service_type == SERVICE_TYPE_DATACITE and doi.doi:
+                        method = WEB_METHOD_PUT
+                        url = '{url}/{doi}'.format(
+                            url=self._config.get('DATACITE', 'url'), doi=doi.doi
+                        )
+                    # Otherwise, for both DataCite and OSTI, just a POST request
+                    # on the default endpoint is sufficient
+                    else:
+                        method = WEB_METHOD_POST
+                        url = self._config.get(service_type.upper(), 'url')
+
```

Comment:
If we ever have a _third_ kind of provider (DataCite, OSTI, and Joe's Citation and 1 Hour Lube Service, for example), we may want to replace the `if/else` and OO'ify this with polymorphism that "does the right thing" based on the kind of service type.

Is there a chance of a third kind of provider?

Reply:
@nutjob4life anything is possible, but highly unlikely